src/modules/types.rs

```diff
@@ -14,14 +14,26 @@ pub enum Type {
     Generic
 }
 
+impl Type {
+    pub fn is_subset_of(&self, of: &Type) -> bool {
+        match (self, of) {
+            (Type::Array(l), Type::Array(r)) => {
```

Comment:
i meant that the variables are way unreadable, there is no reason to not name them `left` and `right`